### PR TITLE
Issue/26

### LINKED ISSUE #26: Input validation in builder
Need to do validation in build() functions for all builders.
- Check type
- Check if undefined is acceptable

### PR DESCRIPTION
**Pull request by scalarindetail**
Fixed #26

### CODE REVIEW

File: request/builder.js [hunk 1]
```diff
@@ -18,6 +18,22 @@ class TextEncoder {
...
+/**
+ * An internal class to validate input
+ * @class
+ */
+class Validator {
+  /**
+   * @param input
+   * @param type
+   */
+  inputValidation(input, type) {
```
Comment (scalarindetail) [inline request/builder.js]:
```suggestion
  validateInput(input, type) {
```

File: request/builder.js [hunk 2]
```diff
@@ -18,6 +18,22 @@ class TextEncoder {
...
 }
 
+/**
+ * An internal class to validate input
+ * @class
+ */
+class Validator {
+  /**
+   * @param input
+   * @param type
```
Comment (scalarindetail) [inline request/builder.js]:
```suggestion
   * @param {*} input
   * @param {Object} type
```

File: request/builder.js [hunk 3]
```diff
@@ -66,8 +82,15 @@ class CertificateRegistrationRequestBuilder {
   /**
    * Builds the CertificateRegistrationRequest
    * @return {CertificateRegistrationRequest}
+   * @throws {Error}
    */
   async build() {
+    const validator = new Validator();
+    validator.validateInput(this.certHolderId, String);
+    validator.validateInput(this.certVersion, Number);
+    validator.validateInput(this.certPem, String);
```
Comment (feeblefakie) [inline request/builder.js]: What will  happen if `certPem` is null ?

Comment (supl) [reply request/builder.js]: Sorry, I thought we are still using `typeof` to test type.
In the way of `input.constructor === type`, we have to check if input is `null` or `undefined`.

File: request/builder.js [hunk 4]
```diff
@@ -66,8 +87,15 @@ class CertificateRegistrationRequestBuilder {
...
+
```
Comment (feeblefakie) [inline request/builder.js]: Please  remove a blank llne. (since doubled blank lines)

Comment (feeblefakie) [reply request/builder.js]: @penpyt Please fix it the it's good to go.

Comment (Torch3333) [reply request/builder.js]: @feeblefakie 
I removed it.

**Pull request merged**